### DISCUSSION TITLE
How to get the  local position of a client position on canvas within a container/displayable object?

### DISCUSSION THREAD

**Discussion by lanyusan** [pixijs/pixijs#8012] [Q&A] [answered]
We are working on a prototype with pixijs as the base lib for a drawing project. We are quite new to Pixi so this might be a naive question for the community.

We have got a use case that needs to translate a client position as defined in the doc  [PIXI.FederatedPointerEvent](https://pixijs.download/dev/docs/PIXI.FederatedPointerEvent.html#client) : `The coordinates of the mouse event relative to the canvas.` to container / displayable object local position. 

I couldn't  figured out an easy way to do the translation as the container / displayable object could be multi layers down a container tree.

Is there an API to do so? I understand there was this method of `getLocalPosition (displayObject, point, globalPos)` of `PIXI.InteractionData`. But it seems to have been deprecated.

Thanks in advance.

**Comment by eXponenta** [comment, marked as answer]
Event has global (pixi world space) position

https://pixijs.download/dev/docs/PIXI.FederatedPointerEvent.html#global

And you can transform it by toLocal 

https://pixijs.download/dev/docs/PIXI.Container.html#toLocal

```

function handle(event) {
     const local = container.toLocal(event.global);
}
```

Plz, use a docs.

**Reply by lanyusan** [reply]
Thanks.
